src/main/webapp/grocery.html

```diff
@@ -12,7 +12,7 @@
                 <button class="dropbtn">MENU</button>
                 <div class="dropdown-content">
                     <a href="home.html">Settings</a>
-                    <a href="cal.html">Calendar</a>
+                    <a href="/calendar-auth">Calendar</a>
```

Comment:
maybe a naive question, but why /calendar-auth instead of just /calendar?

Reply:
I dont actually think this is necessary, but I was running into an issue where the /calendar servlet code was running before the authorization flow happened, so I decided to make a servlet dedicated to initializing the authorization first which then redirects to cal and then /calendar is called

Reply:
This can still be changed in the future